### PR TITLE
Refactor: music/commands and added pagination buttons

### PR DESCRIPTION
**Pull request by 1Kuroyasha**
- Converted the rest of the music commands to slash commands
- Replaced reactions with buttons for the paginated messages

### CODE REVIEW

File: src/listeners/InteractionListener.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ export default class InteractionListener extends Listener {
 	onInteractionCreate = async (interaction: CommandInteraction) => {
 		try {
 			const { commandName } = interaction;
+			if (commandName === undefined) return;
```
Comment (KL13NT) [inline src/listeners/InteractionListener.ts:16]: Use general `!commandName` instead.

File: src/structures/PaginatedEmbed.ts [hunk 2]
```diff
@@ -1,33 +1,51 @@
...
 		"⏪": () => 0,
 		"◀️": () => (this.page > 0 ? this.page - 1 : this.page),
 		"▶️": () => (this.page < this.pages.length - 1 ? this.page + 1 : this.page),
 		"⏩": () => this.pages.length - 1,
 	};
 
-	current: Message;
-	page: number;
-	timeout: number;
+	buttonDisableCondition = {
```
Comment (KL13NT) [inline src/structures/PaginatedEmbed.ts:34]: conditions*

File: src/structures/PaginatedEmbed.ts [hunk 3]
```diff
@@ -36,7 +54,20 @@ export default class PaginationEmbed {
...
-		this.current = await this.channel.send({
+		const buttonRow: Row = {
+			type: "ACTION_ROW",
+			components: Object.keys(this.emojis).map(emoji => {
+				return {
+					custom_id: emoji,
+					emoji: emoji,
+					style: ButtonStyle.Primary,
+					type: "BUTTON",
+					disabled: this.buttonDisableCondition[emoji](),
```
Comment (KL13NT) [inline src/structures/PaginatedEmbed.ts:65]: Move this to the `map` scope as a constant.

File: src/structures/PaginatedEmbed.ts [hunk 4]
```diff
@@ -45,33 +76,43 @@ export default class PaginationEmbed {
...
-			Object.keys(this.emojis).includes(reaction.emoji.name) &&
-			!user.bot &&
-			user.id === this.member.id;
+		const filter = (interaction: MessageComponentInteraction) =>
+			interaction.isButton &&
+			Object.keys(this.emojis).includes(interaction.customId) &&
+			interaction.user.id === this.member.id;
 
-		const collector = this.current.createReactionCollector({
+		const collector = this.interaction.channel.createMessageComponentCollector({
```
Comment (KL13NT) [inline src/structures/PaginatedEmbed.ts:89]: `this.channel`

File: src/structures/PaginatedEmbed.ts [hunk 5]
```diff
@@ -45,33 +76,43 @@ export default class PaginationEmbed {
...
 	};
 
-	onReaction = async (reaction: MessageReaction) => {
-		const modifier = this.emojis[reaction.emoji.name];
+	onButtonClick = async (interaction: MessageComponentInteraction) => {
+		const modifier = this.emojis[interaction.customId];
 		this.page = modifier();
 
-		await this.current.edit({
+		const buttonRow: Row = {
```
Comment (KL13NT) [inline src/structures/PaginatedEmbed.ts:102]: Move all `row` generating code to a private method `generateRow`.

**Review by KL13NT — CHANGES_REQUESTED** [review]
Well done 💖 
Just a few tiny modifications and you're good to go💀

File: src/structures/PaginatedEmbed.ts [hunk 6]
```diff
@@ -53,19 +53,25 @@ export default class PaginationEmbed {
 		this.timeout = timeout;
 	}
 
-	init = async () => {
-		const buttonRow: Row = {
+	generateRow(): Row {
```
Comment (KL13NT) [inline src/structures/PaginatedEmbed.ts:56]: Arrow

File: src/structures/PaginatedEmbed.ts [hunk 7]
```diff
@@ -53,19 +53,25 @@ export default class PaginationEmbed {
...
 					type: "BUTTON",
-					disabled: this.buttonDisableCondition[emoji](),
+					disabled,
 				};
 			}),
 		};
+	}
+
+	init = async () => {
+		const buttonRow: Row = this.generateRow();
```
Comment (KL13NT) [inline src/structures/PaginatedEmbed.ts:74]: Remove type

File: src/structures/PaginatedEmbed.ts [hunk 8]
```diff
@@ -99,18 +105,7 @@ export default class PaginationEmbed {
...
-				return {
-					custom_id: emoji,
-					emoji: emoji,
-					style: ButtonStyle.Primary,
-					type: "BUTTON",
-					disabled: this.buttonDisableCondition[emoji](),
-				};
-			}),
-		};
+		const buttonRow: Row = this.generateRow();
```
Comment (KL13NT) [inline src/structures/PaginatedEmbed.ts:108]: Remove type

**Review by KL13NT — APPROVED** [review]
Thanks for your valuable contributions! 🥳💖

**Comment by 1Kuroyasha** [comment]
💀💀

**Pull request merged**